### ISSUE TITLE
Handling for field names with "." needs to be fixed

### ISSUE THREAD

**Issue by bena-nasa** [GEOS-ESM/MAPL#731]
The UFS/MAPL/GOCART work exposed a bug in History.
The issue is when you have "." in the field names and put them in a collection:
```
  inst_ca_3d.format:     'CFIO' ,
  inst_ca_3d.descr:      '3d,3-Hourly,Instantaneous,Model-Level,Analysis,Aerosol Concentrations',
  inst_ca_3d.template:   '%y4%m2%d2_%h2%n2z.nc4',
  inst_ca_3d.mode:       'instantaneous',
  inst_ca_3d.frequency:   030000 ,
  inst_ca_3d.fields:   'fooCA.bc' , 'CA.bc'    ,
                      ::
```
The problem is that History interprets this as an expression because of the "."
The causes History to unnecessarily add the field twice to the state. History tries to identify extra fields in the expression that are not already in the collection as it can pull these from the component specified in the expression. The "." in the name meant that "fooCA.bc" was not being scanned for in the other fields in the collection because of the presence of one of these special characters "()+-/*." and hence was not being added to the list of fields already in the collection. Then it thinks that is an expression because it was making this determination based on the presence of one of these special characters "()+-/*.". Technically this is an expression in that the name of a single field is a perfectly valid expression, it is just treated as an identity by the parser, if the expression is "fooCA.bc" and "fooCA.bc" is a field in the state delivered to the parser it fill the output field with "fooCA.bc". The bottom line, there is a routine that returns a list of fields that in this case is just "fooCA.bc" and adds it to a state. This should not have happened in the first place.

 Then history loops over the actual  list of fields in the collection and adds those to this same state. The name of the field added is actual the alias, which if none is provided is the same as the short name. Expressions must have aliases. 

At this point "fooCA.bc" is added to this state, which already as "fooCA.bc" and is an error!

So the routine that parses the expression should be smarter. It should first do a check, is "fooCA.bc" in the component named in entry for that field. If that is the case, we are done, it is not an expression and it should be added to the list of fields already in the collection. I will make the appropriate PR to fix this.

**Comment by rmontuoro** [comment]
PR #734 seems to solve this issue, but further testing is needed once the current issues with MAPL `develop` are fixed (#737 )

**Comment by bena-nasa** [comment]
Closing this. The issue is fixed on develop.
